<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>ModuleEditor</class>
 <widget class="QMainWindow" name="ModuleEditor">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>640</width>
    <height>480</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Module Editor</string>
  </property>
  <widget class="QWidget" name="centralWidget">
   <layout class="QFormLayout" name="formLayout">
    <property name="fieldGrowthPolicy">
     <enum>QFormLayout::AllNonFixedFieldsGrow</enum>
    </property>
    <item row="0" column="0">
     <widget class="QLabel" name="label_4">
      <property name="text">
       <string>Module Name</string>
      </property>
     </widget>
    </item>
    <item row="0" column="1">
     <widget class="QLineEdit" name="moduleNameText"/>
    </item>
    <item row="1" column="0">
     <widget class="QLabel" name="label_3">
      <property name="text">
       <string>Description</string>
      </property>
     </widget>
    </item>
    <item row="2" column="0">
     <widget class="QLabel" name="label">
      <property name="text">
       <string>Domain</string>
      </property>
     </widget>
    </item>
    <item row="6" column="1">
     <widget class="QLineEdit" name="codeLocationText"/>
    </item>
    <item row="5" column="1">
     <widget class="QLineEdit" name="classNameText"/>
    </item>
    <item row="5" column="0">
     <widget class="QLabel" name="label_2">
      <property name="text">
       <string>Class Name</string>
      </property>
     </widget>
    </item>
    <item row="6" column="0">
     <widget class="QLabel" name="label_5">
      <property name="text">
       <string>Code Location</string>
      </property>
     </widget>
    </item>
    <item row="7" column="1">
     <widget class="QCheckBox" name="parallelCheck">
      <property name="text">
       <string>Parallel Execution Support</string>
      </property>
     </widget>
    </item>
    <item row="2" column="1">
     <widget class="QComboBox" name="domainCombo"/>
    </item>
    <item row="4" column="0">
     <widget class="QLabel" name="label_6">
      <property name="text">
       <string>Stopping Conditions</string>
      </property>
     </widget>
    </item>
    <item row="4" column="1">
     <layout class="QGridLayout" name="stopcondGrid"/>
    </item>
    <item row="1" column="1">
     <widget class="QPlainTextEdit" name="descriptionText">
      <property name="tabChangesFocus">
       <bool>true</bool>
      </property>
      <property name="backgroundVisible">
       <bool>false</bool>
      </property>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>640</width>
     <height>20</height>
    </rect>
   </property>
   <widget class="QMenu" name="menuFile">
    <property name="title">
     <string>File</string>
    </property>
    <addaction name="actionNew"/>
    <addaction name="actionOpen"/>
    <addaction name="separator"/>
    <addaction name="actionSave"/>
    <addaction name="actionSave_As"/>
    <addaction name="separator"/>
    <addaction name="actionClose"/>
   </widget>
   <addaction name="menuFile"/>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
  <action name="actionOpen">
   <property name="text">
    <string>Open...</string>
   </property>
   <property name="toolTip">
    <string>Open a module XML file.</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+O</string>
   </property>
  </action>
  <action name="actionSave">
   <property name="text">
    <string>Save</string>
   </property>
   <property name="toolTip">
    <string>Saves the current module to XML.</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+S</string>
   </property>
  </action>
  <action name="actionClose">
   <property name="text">
    <string>Close</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+W</string>
   </property>
  </action>
  <action name="actionSave_As">
   <property name="text">
    <string>Save As...</string>
   </property>
   <property name="toolTip">
    <string>Saves the current module to a new XML file.</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+Shift+S</string>
   </property>
  </action>
  <action name="actionNew">
   <property name="text">
    <string>New...</string>
   </property>
   <property name="toolTip">
    <string>New...</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+N</string>
   </property>
  </action>
 </widget>
 <tabstops>
  <tabstop>moduleNameText</tabstop>
  <tabstop>descriptionText</tabstop>
  <tabstop>domainCombo</tabstop>
  <tabstop>classNameText</tabstop>
  <tabstop>codeLocationText</tabstop>
  <tabstop>parallelCheck</tabstop>
 </tabstops>
 <resources/>
 <connections>
  <connection>
   <sender>actionClose</sender>
   <signal>triggered()</signal>
   <receiver>ModuleEditor</receiver>
   <slot>close()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>399</x>
     <y>299</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionOpen</sender>
   <signal>triggered()</signal>
   <receiver>ModuleEditor</receiver>
   <slot>openModule()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>319</x>
     <y>239</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionSave</sender>
   <signal>triggered()</signal>
   <receiver>ModuleEditor</receiver>
   <slot>saveModule()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>319</x>
     <y>239</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>moduleNameText</sender>
   <signal>textChanged(QString)</signal>
   <receiver>ModuleEditor</receiver>
   <slot>nameChanged(QString)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>391</x>
     <y>47</y>
    </hint>
    <hint type="destinationlabel">
     <x>319</x>
     <y>239</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>classNameText</sender>
   <signal>textChanged(QString)</signal>
   <receiver>ModuleEditor</receiver>
   <slot>classNameChanged(QString)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>391</x>
     <y>340</y>
    </hint>
    <hint type="destinationlabel">
     <x>319</x>
     <y>239</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>codeLocationText</sender>
   <signal>textChanged(QString)</signal>
   <receiver>ModuleEditor</receiver>
   <slot>codeLocationChanged(QString)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>391</x>
     <y>373</y>
    </hint>
    <hint type="destinationlabel">
     <x>319</x>
     <y>239</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>parallelCheck</sender>
   <signal>toggled(bool)</signal>
   <receiver>ModuleEditor</receiver>
   <slot>isParallelChanged(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>391</x>
     <y>403</y>
    </hint>
    <hint type="destinationlabel">
     <x>319</x>
     <y>239</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionSave_As</sender>
   <signal>triggered()</signal>
   <receiver>ModuleEditor</receiver>
   <slot>saveModuleAs()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>319</x>
     <y>239</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionNew</sender>
   <signal>triggered()</signal>
   <receiver>ModuleEditor</receiver>
   <slot>newModule()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>319</x>
     <y>239</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>domainCombo</sender>
   <signal>currentIndexChanged(QString)</signal>
   <receiver>ModuleEditor</receiver>
   <slot>domainChanged(QString)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>378</x>
     <y>257</y>
    </hint>
    <hint type="destinationlabel">
     <x>319</x>
     <y>239</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>descriptionText</sender>
   <signal>textChanged()</signal>
   <receiver>ModuleEditor</receiver>
   <slot>descriptionChanged()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>378</x>
     <y>145</y>
    </hint>
    <hint type="destinationlabel">
     <x>319</x>
     <y>239</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <slot>openModule()</slot>
  <slot>saveModule()</slot>
  <slot>descriptionChanged()</slot>
  <slot>nameChanged(QString)</slot>
  <slot>classNameChanged(QString)</slot>
  <slot>codeLocationChanged(QString)</slot>
  <slot>isParallelChanged(bool)</slot>
  <slot>stopCondChanged()</slot>
  <slot>saveModuleAs()</slot>
  <slot>newModule()</slot>
  <slot>domainChanged(QString)</slot>
 </slots>
</ui>
